# Notebook 1: Exploring the corpus
Shannon Wietsma - S4147472

In [30]:
import pandas as pd

In [32]:
corpus = pd.read_csv("corpus.csv", delimiter = ";", encoding = 'utf8')
corpus.sample(5)

blog_name                     title   
1295      creativepromptsforwriting  Creative Writing Prompts  \
893                writing-prompt-s           Writing Prompts   
452                writing-prompt-s           Writing Prompts   
2245  promptsforthestrugglingauthor     Writing Prompts Daily   
1591      creativepromptsforwriting  Creative Writing Prompts   

                                            description   
1295  Writing Sideblog by <a href="https://agirlname...  \
893   I am Prompt Guy and you are the protagonist. L...   
452   I am Prompt Guy and you are the protagonist. L...   
2245  <a href=https://promptsforthestrugglingauthor....   
1591  Writing Sideblog by <a href="https://agirlname...   

                                               post_url date (YYYY-MM-DD)   
1295  https://creativepromptsforwriting.tumblr.com/p...         11-3-2023  \
893   https://writing-prompt-s.tumblr.com/post/71026...         25-2-2023   
452   https://writing-prompt-s.tumblr.com/post/71134...          9-3-2023   
2245  https://www.tumblr.com/blog/view/promptsforthe...          3-3-2023   
1591  https://creativepromptsforwriting.tumblr.com/p...          6-3-2023   

                                               hashtags   
1295  ['creativepromptsforwriting', 'write', 'writin...  \
893   ['writers', 'writers on tumblr', 'writing prom...   
452   ['writers', 'writers on tumblr', 'writing prom...   
2245                                                NaN   
1591  ["let's create a character", 'create a charact...   

                              short_url writing_prompt_id   
1295  https://tmblr.co/Z_s_RjdVuwi0am00             wp156  \
893   https://tmblr.co/ZCDX7hdRNSX00W00             wp045   
452   https://tmblr.co/ZCDX7hdVDSBkim00             wp023   
2245  https://tmblr.co/ZnM30fdT0X_GOu00             wp226   
1591  https://tmblr.co/Z_s_RjdU7mEHKe00             wp174   

                                         writing_prompt  total_amount_notes   
1295  <p><a class="tumblr_blog" href="https://creati...                 819  \
893   <p>“IF NOT FRIEND, THEN WHY FRIEND SHAPED!?” t...               16398   
452   <p>The lottery is a secret plot to catch time ...                3564   
2245  <p><a class="tumblr_blog" href="https://www.tu...                 551   
1591  <h1><b>Let&rsquo;s create a character!</b></h1...                 190   

     response_id                                           response   
1295   wp156r016       #saving for later #creativepromptsforwriting  \
893    wp045r014  #writers #writers on tumblr #writing prompts #...   
452    wp023r013                                   #writing prompts   
2245   wp226r005  #writing #writing prompt #writing prompts #pro...   
1591   wp174r009  #the names are from Turkish! #chose ümit becau...   

            blog_name_response type_of_response   
1295  deaththehighcostofliving           reblog  \
893             newdawnhorizon           reblog   
452               starfishride           reblog   
2245          seaside-writings           reblog   
1591            thewriterghost           reblog   

                  reblog_parent_name  
1295       creativepromptsforwriting  
893                 writing-prompt-s  
452                 writing-prompt-s  
2245  writing-challenges-and-prompts  
1591       creativepromptsforwriting

# The basics to explore the dataset

In [33]:
# To show the column names
corpus.columns

Index(['blog_name', 'title', 'description', 'post_url', 'date (YYYY-MM-DD)',
       'hashtags', 'short_url', 'writing_prompt_id', 'writing_prompt',
       'total_amount_notes', 'response_id', 'response', 'blog_name_response',
       'type_of_response', 'reblog_parent_name'],
      dtype='object')

In [34]:
# To show a concise summary of the dataset
corpus.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   blog_name           2500 non-null   object
 1   title               2500 non-null   object
 2   description         2500 non-null   object
 3   post_url            2500 non-null   object
 4   date (YYYY-MM-DD)   2500 non-null   object
 5   hashtags            2133 non-null   object
 6   short_url           2500 non-null   object
 7   writing_prompt_id   2500 non-null   object
 8   writing_prompt      2500 non-null   object
 9   total_amount_notes  2500 non-null   int64 
 10  response_id         2500 non-null   object
 11  response            2467 non-null   object
 12  blog_name_response  2468 non-null   object
 13  type_of_response    2467 non-null   object
 14  reblog_parent_name  1787 non-null   object
dtypes: int64(1), object(14)
memory usage: 293.1+ KB


In [35]:
# To get summary statistics of the dataset. 
# As becomes visible, only total_amount_notes is a numerical variable. 
corpus.describe()

total_amount_notes
count          2500.00000
mean           3426.87560
std            7709.39656
min               2.00000
25%             230.00000
50%             558.00000
75%            4364.00000
max           64992.00000

In [36]:
corpus['writing_prompt_id'].value_counts()


writing_prompt_id
wp001    20
wp204    20
wp192    20
wp193    20
wp195    20
         ..
wp181     1
wp124     1
wp128     1
wp129     1
wp126     1
Name: count, Length: 250, dtype: int64

In [37]:
print("Short summary of the writing prompts")
corpus['writing_prompt_id'].describe()

Short summary of the writing prompts


count      2500
unique      250
top       wp001
freq         20
Name: writing_prompt_id, dtype: object

In [38]:
print("Short summary of the responses")
corpus['response_id'].describe()

Short summary of the responses


count          2500
unique         2500
top       wp001r001
freq              1
Name: response_id, dtype: object

In [39]:
print("Blog name with the amount of responses")
print(corpus['blog_name'].value_counts())

Blog name with the amount of responses
blog_name
writing-prompt-s                 1000
creativepromptsforwriting         767
promptsforthestrugglingauthor     515
writingprompts                    142
dropkickwritersblock               76
Name: count, dtype: int64


In [40]:
print("Those interacting with the post, in terms of frequency")
print(corpus['blog_name_response'].value_counts())

Those interacting with the post, in terms of frequency
blog_name_response
newdawnhorizon                    152
local-single-wizard                49
myseconddumpsterfire               27
writing-challenges-and-prompts     20
cocoamoonmalfoy                    13
                                 ... 
cleolinda                           1
mixmangosmangoverse                 1
nwaml                               1
soibeanxgalactica                   1
geldar13                            1
Name: count, Length: 1554, dtype: int64


### Deleting writing prompts, and corresponding responses that are disqualified as writing prompts. 

- 30 (reblog of a joke)
- 35 (reblog of a recipe)
- 44 (crowdfund idea)
- 52 (reflection on CHATGPT in combination with prompts)
- 137 (inquiry of preferences audience)
- 144 (post about 2 week break)
- 151 (poll about language)
- 153 (submission provides advice)
- 159 (answers what ‘whump’ is)
- 164 (petitions for more characters with glasses)
- 165 (explains what ‘canon’ vs ‘fanon’ is)
- 169 (writing game)
- 170 (writing game)
- 171 (writing game)
- 172 (writing game)
- 174 (writing game)
- 178 (writing game)
- 179 (prompts only available for certain Tumblr users)
- 181 (apologizes for absence of answer in submission)
- 182 (apologizes for absence of answer in submission)
- 183 (writing game)
- 184 (writing game)
- 186 (writing game)
- 187 (explains what ‘plot bunny’ is)
- 188 (writing game)
- 190 (shout-out)
- 192 (writing game)
- 195 (writing game)
- 197 (more affirmation that people are writers)
- 198 (poll about preferences related to lovers)
- 200 (poll about preferences related to lovers)

In [41]:
# The below is code to delete the prompts that are not considered as actual writing prompts
clean_corpus = corpus
clean_corpus
 
numbers_to_delete = ['wp030', 'wp035', 'wp044', 'wp052', 'wp137', 'wp144', 'wp151', 'wp153', 'wp159', 'wp164', 'wp165', 'wp169', 'wp170', 'wp171', 'wp172', 'wp174', 'wp178', 'wp179', 'wp181', 'wp182', 'wp183', 'wp184', 'wp186', 'wp187', 'wp188', 'wp190', 'wp192', 'wp195', 'wp197', 'wp198', 'wp200']

clean_corpus = clean_corpus[~clean_corpus['writing_prompt_id'].isin(numbers_to_delete)]
clean_corpus

blog_name                  title   
0                  writing-prompt-s        Writing Prompts  \
1                  writing-prompt-s        Writing Prompts   
2                  writing-prompt-s        Writing Prompts   
3                  writing-prompt-s        Writing Prompts   
4                  writing-prompt-s        Writing Prompts   
...                             ...                    ...   
2495  promptsforthestrugglingauthor  Writing Prompts Daily   
2496  promptsforthestrugglingauthor  Writing Prompts Daily   
2497  promptsforthestrugglingauthor  Writing Prompts Daily   
2498  promptsforthestrugglingauthor  Writing Prompts Daily   
2499  promptsforthestrugglingauthor  Writing Prompts Daily   

                                            description   
0     I am Prompt Guy and you are the protagonist. L...  \
1     I am Prompt Guy and you are the protagonist. L...   
2     I am Prompt Guy and you are the protagonist. L...   
3     I am Prompt Guy and you are the protagonist. L...   
4     I am Prompt Guy and you are the protagonist. L...   
...                                                 ...   
2495  <a href=https://promptsforthestrugglingauthor....   
2496  <a href=https://promptsforthestrugglingauthor....   
2497  <a href=https://promptsforthestrugglingauthor....   
2498  <a href=https://promptsforthestrugglingauthor....   
2499  <a href=https://promptsforthestrugglingauthor....   

                                               post_url date (YYYY-MM-DD)   
0     https://writing-prompt-s.tumblr.com/post/71185...         15-3-2023  \
1     https://writing-prompt-s.tumblr.com/post/71185...         15-3-2023   
2     https://writing-prompt-s.tumblr.com/post/71185...         15-3-2023   
3     https://writing-prompt-s.tumblr.com/post/71185...         15-3-2023   
4     https://writing-prompt-s.tumblr.com/post/71185...         15-3-2023   
...                                                 ...               ...   
2495  https://www.tumblr.com/blog/view/promptsforthe...         18-2-2023   
2496  https://www.tumblr.com/blog/view/promptsforthe...         18-2-2023   
2497  https://www.tumblr.com/blog/view/promptsforthe...         18-2-2023   
2498  https://www.tumblr.com/blog/view/promptsforthe...         18-2-2023   
2499  https://www.tumblr.com/blog/view/promptsforthe...         18-2-2023   

                                               hashtags   
0                                                   NaN  \
1                                                   NaN   
2                                                   NaN   
3                                                   NaN   
4                                                   NaN   
...                                                 ...   
2495  ['writing prompt', 'writing', 'writers on tumb...   
2496  ['writing prompt', 'writing', 'writers on tumb...   
2497  ['writing prompt', 'writing', 'writers on tumb...   
2498  ['writing prompt', 'writing', 'writers on tumb...   
2499  ['writing prompt', 'writing', 'writers on tumb...   

                              short_url writing_prompt_id   
0     https://tmblr.co/ZCDX7hdX0UumSa00             wp001  \
1     https://tmblr.co/ZCDX7hdX0UumSa00             wp001   
2     https://tmblr.co/ZCDX7hdX0UumSa00             wp001   
3     https://tmblr.co/ZCDX7hdX0UumSa00             wp001   
4     https://tmblr.co/ZCDX7hdX0UumSa00             wp001   
...                                 ...               ...   
2495  https://tmblr.co/ZnM30fdP3TYvGm00             wp250   
2496  https://tmblr.co/ZnM30fdP3TYvGm00             wp250   
2497  https://tmblr.co/ZnM30fdP3TYvGm00             wp250   
2498  https://tmblr.co/ZnM30fdP3TYvGm00             wp250   
2499  https://tmblr.co/ZnM30fdP3TYvGm00             wp250   

                                         writing_prompt  total_amount_notes   
0     <p>You’ve accidentally summoned the Devil whil...                4245  \
1     <p>You’ve accidentally summoned the Devil

# Checking whether there are NA entries for the 'response' column. 

In [42]:
# Check which are NA entries for the 'response' column
empty_cells = clean_corpus['response'].isnull()
empty_indices = empty_cells[empty_cells].index
empty_indices

Index([1100, 1101, 1113, 1116, 1118, 1123, 1125, 1127, 1129, 1130, 1133, 1134,
       1135, 1140, 1144, 1181, 1193, 1196, 1197, 1198, 1201, 1209, 1210, 1212,
       1213, 1214, 1215, 1216, 1217, 1341],
      dtype='int64')

In [43]:
empty_indices = [1100, 1101, 1113, 1116, 1118, 1123, 1125, 1127, 1129, 1130, 1133, 1134,
                 1135, 1140, 1144, 1181, 1193, 1196, 1197, 1198, 1201, 1209, 1210, 1212,
                 1213, 1214, 1215, 1216, 1217, 1341]

# Reset the index of the clean_corpus DataFrame
clean_corpus_reset = clean_corpus.reset_index(drop=True)

# Check if the indices exist in the DataFrame's index
valid_indices = [idx for idx in empty_indices if idx < len(clean_corpus_reset)]

# Retrieve the response_id values corresponding to the valid indices
response_ids = clean_corpus_reset.loc[valid_indices, 'response_id']

# Print the response_id values
print(response_ids)



1100    wp114r001
1101    wp115r001
1113    wp122r001
1116    wp125r001
1118    wp125r003
1123    wp127r004
1125    wp128r001
1127    wp130r001
1129    wp131r001
1130    wp132r001
1133    wp135r001
1134    wp136r001
1135    wp138r001
1140    wp141r001
1144    wp146r001
1181    wp155r012
1193    wp156r004
1196    wp156r007
1197    wp156r008
1198    wp156r009
1201    wp156r012
1209    wp156r020
1210    wp157r001
1212    wp157r003
1213    wp157r004
1214    wp157r005
1215    wp157r006
1216    wp157r007
1217    wp157r008
1341    wp167r018
Name: response_id, dtype: object


In [14]:
response_ids = [
    'wp125r001',
    'wp125r002',
    'wp135r001',
    'wp138r003',
    'wp140r001',
    'wp152r004',
    'wp152r006',
    'wp152r008',
    'wp152r010',
    'wp152r011',
    'wp152r014',
    'wp152r015',
    'wp152r016',
    'wp154r001',
    'wp155r004',
    'wp157r001',
    'wp157r013',
    'wp157r016',
    'wp157r017',
    'wp157r018',
    'wp158r001',
    'wp158r009',
    'wp158r010',
    'wp158r012',
    'wp158r013',
    'wp158r014',
    'wp158r015',
    'wp158r016',
    'wp158r017',
    'wp173r008'
]

# Filter the clean_corpus DataFrame based on response_id values
filtered_corpus = clean_corpus[clean_corpus['response_id'].isin(response_ids)]

# Print the writing_prompts and responses
for _, row in filtered_corpus.iterrows():
    print("Writing Prompt:", row['writing_prompt'])
    print("Response:", row['response'])
    print()

#This shows that all the data thought to be empty in the 'response' column, is actually not empty


Writing Prompt: <p>Write a piece about something small that grows ten thousand times its original size</p>
Response: Occamies? Hmm. ??

Writing Prompt: <p>Write a piece about something small that grows ten thousand times its original size</p>
Response: DO IT!!! I’ll be waiting. I’ll be the first to read it.

Writing Prompt: <p>Write a piece about a missed chance</p>
Response: #writing prompts

Writing Prompt: <p>Write a piece about pure bliss</p>
Response: #insp #prompts #Eurdyn & Aurelian

Writing Prompt: <p>Write a piece about someone who cannot follow instructions</p>
Response: No I don't think I will

Writing Prompt: 'asking_name': 'Anonymous', 'asking_url': None, 'question': '<p>heyy i saw that in your AU ideas list you have Rockstar AU but no prompts for that, so i was wondering if you could come up with some? especially for a rockstar dating an ‘ordinary’ person, not involved in their world. thank you! </p>', 'answer': '<p>Hi :)</p><p>You&rsquo;re right, there are no actual Rock

# Summaries of the data

In [15]:
print("Short summary of the writing prompts")
clean_corpus['writing_prompt_id'].describe()

Short summary of the writing prompts


count      2099
unique      219
top       wp001
freq         20
Name: writing_prompt_id, dtype: object

In [16]:
print("Short summary of the responses")
clean_corpus['response_id'].describe()

Short summary of the responses


count          2099
unique         2099
top       wp001r001
freq              1
Name: response_id, dtype: object

In [17]:
print("Blog name with the amount of responses")
print(clean_corpus['blog_name'].value_counts())

Blog name with the amount of responses
blog_name
writing-prompt-s                 940
promptsforthestrugglingauthor    515
creativepromptsforwriting        435
writingprompts                   135
dropkickwritersblock              74
Name: count, dtype: int64


In [19]:
print("Those interacting with the post, in terms of frequency")
print(clean_corpus['blog_name_response'].value_counts())

Those interacting with the post, in terms of frequency
blog_name_response
newdawnhorizon                    146
local-single-wizard                49
myseconddumpsterfire               24
writing-challenges-and-prompts     20
cocoamoonmalfoy                    13
                                 ... 
thedoctorknits                      1
mygillis                            1
nyenke                              1
lori-kabori                         1
i-am-the-one-who-wololoes           1
Name: count, Length: 1309, dtype: int64


In [ ]:
clean_corpus.to_excel('clean_corpus.xlsx', index = False)
clean_corpus.to_csv('clean_corpus.csv', index = False)

# Clean the analysis for the prompts

In [44]:
# The below is code to clean the analysis for the prompts
content_prompt_df = pd.read_excel("prompts_analysis.xlsx")

numbers_to_delete = ['wp030', 'wp035', 'wp044', 'wp052', 'wp137', 'wp144', 'wp151', 'wp153', 'wp159', 'wp164', 'wp165', 'wp169', 'wp170', 'wp171', 'wp172', 'wp174', 'wp178', 'wp179', 'wp181', 'wp182', 'wp183', 'wp184', 'wp186', 'wp187', 'wp188', 'wp190', 'wp192', 'wp195', 'wp197', 'wp198', 'wp200']

clean_content_prompt = content_prompt_df[~content_prompt_df['writing_prompt_id'].isin(numbers_to_delete)]
clean_content_prompt

writing_prompt_id reblog submission original content   
0               wp001     no         no              yes  \
1               wp002     no         no              yes   
2               wp003     no         no              yes   
3               wp004     no         no              yes   
4               wp005     no         no              yes   
..                ...    ...        ...              ...   
245             wp246     no         no              yes   
246             wp247    yes         no               no   
247             wp248     no         no              yes   
248             wp249    yes         no               no   
249             wp250     no         no              yes   

                                      structure   
0                                       <p></p>  \
1                                       <p></p>   
2                                       <p></p>   
3                                       <p></p>   
4                                       <p></p>   
..                                          ...   
245  <p><h1>Writing Prompt #XXX</h1></p><p></p>   
246                        self-reblog of wp248   
247  <p><h1>Writing Prompt #XXX</h1></p><p></p>   
248                        self-reblog of wp250   
249         <h1>Writing Prompt #XXX</h1><p></p>   

                                              subtheme prompt framing   
0    Fantasy and the Supernatural, Mystery and Intr...    description  \
1    Power and Control, Fanasy and the Supernatural...    combination   
2    Love and Relationships, Alternate Realities or...    description   
3    Good vs. Evil, Redemption and Second Chances ,...    description   
4    Discovery and Exploration, Power of Imaginatio...    description   
..                                                 ...            ...   
245  Good vs. Evil, Challenge and Redemption, Coura...   conversation   
246                             Love and Relationships   conversation   
247                             Love and Relationships   conversation   
248  Identity and Self-Discovery, Conformity and In...    description   
249  Identity and Self-Discovery, Conformity and In...    description   

                                             reference reader identification  
0    you, Devil, culinary skills, high-stakes cook-...                   you  
1     vampire, vampire's daughters, you, fight, weapon                   you  
2                     human society, vanishings, herds              implicit  
3    supervillain/you, henchman, sidekick of archne...                   you  
4                          you, handheld device, sleep                   you  
..                                                 ...                   ...  
245                       you, hero routine, sacrifice              implicit  
246                                      you, me, love              implicit  
247                                      you, me, love              implicit  
248                              undead, I, modern era              implicit  
249                              undead, I, modern era              implicit  

[219 rows x 9 columns]

In [23]:
clean_content_prompt.to_excel('clean_content_prompt.xlsx', index = False)
clean_content_prompt.to_csv('clean_content_prompt.csv', index = False)

In [24]:
# seperating clean_content_prompt.csv into 2
# 1: for layer 2 will become 
# 2: for layer 3: writing prompts

In [28]:
# Split clean_content_prompt.csv into the variables necessary for layer 2
for_layer_2 = pd.read_csv('clean_content_prompt.csv')
for_layer_2 = for_layer_2.drop(['subtheme', 'prompt framing', 'reference',
       'reader identification'], axis = 1)
for_layer_2.to_csv('analysis_for_layer_2_to_be_added.csv')
for_layer_2

writing_prompt_id reblog submission original content   
0               wp001     no         no              yes  \
1               wp002     no         no              yes   
2               wp003     no         no              yes   
3               wp004     no         no              yes   
4               wp005     no         no              yes   
..                ...    ...        ...              ...   
214             wp246     no         no              yes   
215             wp247    yes         no               no   
216             wp248     no         no              yes   
217             wp249    yes         no               no   
218             wp250     no         no              yes   

                                      structure  
0                                       <p></p>  
1                                       <p></p>  
2                                       <p></p>  
3                                       <p></p>  
4                                       <p></p>  
..                                          ...  
214  <p><h1>Writing Prompt #XXX</h1></p><p></p>  
215                        self-reblog of wp248  
216  <p><h1>Writing Prompt #XXX</h1></p><p></p>  
217                        self-reblog of wp250  
218         <h1>Writing Prompt #XXX</h1><p></p>  

[219 rows x 5 columns]

In [29]:
for_layer_3_prompts = pd.read_csv('clean_content_prompt.csv')
for_layer_3_prompts = for_layer_3_prompts.drop(['reblog', 'submission', 'original content', 'structure'], axis = 1)
for_layer_3_prompts.to_csv('analysis_for_layer_3_prompts_to_be_added.csv')
for_layer_3_prompts

writing_prompt_id                                           subtheme   
0               wp001  Fantasy and the Supernatural, Mystery and Intr...  \
1               wp002  Power and Control, Fanasy and the Supernatural...   
2               wp003  Love and Relationships, Alternate Realities or...   
3               wp004  Good vs. Evil, Redemption and Second Chances ,...   
4               wp005  Discovery and Exploration, Power of Imaginatio...   
..                ...                                                ...   
214             wp246  Good vs. Evil, Challenge and Redemption, Coura...   
215             wp247                             Love and Relationships   
216             wp248                             Love and Relationships   
217             wp249  Identity and Self-Discovery, Conformity and In...   
218             wp250  Identity and Self-Discovery, Conformity and In...   

    prompt framing                                          reference   
0      description  you, Devil, culinary skills, high-stakes cook-...  \
1      combination   vampire, vampire's daughters, you, fight, weapon   
2      description                   human society, vanishings, herds   
3      description  supervillain/you, henchman, sidekick of archne...   
4      description                        you, handheld device, sleep   
..             ...                                                ...   
214   conversation                       you, hero routine, sacrifice   
215   conversation                                      you, me, love   
216   conversation                                      you, me, love   
217    description                              undead, I, modern era   
218    description                              undead, I, modern era   

    reader identification  
0                     you  
1                     you  
2                implicit  
3                     you  
4                     you  
..                    ...  
214              implicit  
215              implicit  
216              implicit  
217              implicit  
218              implicit  

[219 rows x 5 columns]